### ISSUE TITLE
为什么会默认启动conda？

### ISSUE THREAD

**Issue by gqqnbig** [gqqnbig/shine-cluster#118]
作为一名普通使用者，我发现在dagobah，输入 vim /etc/skel/.bashrc ，该文件有

```bash
# enable programmable completion features (you don't need to enable
# this, if it's already enabled in /etc/bash.bashrc and /etc/profile
# sources /etc/bash.bashrc).
if ! shopt -oq posix; then
  if [ -f /usr/share/bash-completion/bash_completion ]; then
    . /usr/share/bash-completion/bash_completion
  elif [ -f /etc/bash_completion ]; then
    . /etc/bash_completion
  fi
fi
source /opt/anaconda3/bin/activate
```

我不是说我喜欢或不喜欢这段话，只是请问管理员，这一段话是怎么来的？经过了何种讨论、记录在哪个issue或PR？

如果是某个管理员偷偷摸摸加的，请把它去掉，谢谢。

**Comment by gqqnbig** [comment]
Please make non-confidential items above board.

**Comment by luoyuqi-lab** [comment]
I use `stat` to check the file's modification, the most recent modify data is 2022-1-21.

![bachrc time](https://user-images.githubusercontent.com/57832734/165685566-3745bd64-51e0-47c6-adb0-550498010e17.png)

And I check the sudoers in dagobah, only @gqqnbig  and @luoyuqi-lab.

I have no idea about other information, what issue, or who coded it.

**Comment by gqqnbig** [comment]
unrecognized code deleted

**Comment by luoyuqi-lab** [comment]
SHA is 4f2de0d239bffedc8b9449b590673b09c8d42a38.

**Comment by gqqnbig** [comment]
https://github.com/gqqnbig/shine-cluster/wiki/GPU%E6%9C%8D%E5%8A%A1%E5%99%A8%E5%AE%89%E8%A3%85/_compare/db321b9edf7a4dcee633d386b74938cadc5d768f...4f2de0d239bffedc8b9449b590673b09c8d42a38

**Comment by gqqnbig** [comment]
@luoyuqi-lab 

https://github.com/gqqnbig/shine-cluster/wiki/GPU%E6%9C%8D%E5%8A%A1%E5%99%A8%E5%AE%89%E8%A3%85 该页面有争议。

把文档写在wiki的好处是：容易更新，不需要code review，commit message空间不太（而写不下理由）。

当wiki页面有争议，我会把该内容放到repo。对repo文件的修改需要code review。

**Comment by gqqnbig** [comment]
blocked by #48

**Comment by gqqnbig** [comment]
```
# 添加 source /opt/anaconda3/bin/activate 到 /etc/skel/.bashrc
```

this line actually is related to #126 because the latter imports anaconda.

Blocked by #126.

**Comment by gqqnbig** [comment]
I figure we should touch less bash config (as well as zsh config).

To begin with, the source statement in .bashrc creates troubles when users try to run sbatch jobs. As a proof, a lot of times users ask me for help when their sbatch files don't have the statement that activates conda but they want to use conda. Users don't know bash implicitly runs .bashrc when they go to a compute node and they thus don't think the source statement is essential in a sbatch job involving conda.

What's more, not shipping the source statement reduces our teaching responsibility. Whoever tells a user to add code to .bashrc for automation benefits has a responsibility to inform the implication for sbatch. After all, our official guide or code doesn't recommend this and since you don't follow our recommended practice, we are not obligated to fix your sbatch problems.

To sum up, while it's ok to give users the source statement for them to paste every time, we should not add the statement to .bashrc for automatic execution for improving user awareness, reducing our responsibility, and cutting down code complexity.

@luoyuqi-lab

**Comment by gqqnbig** [comment]
fixed by https://github.com/gqqnbig/shine-cluster/wiki/%E5%AE%89%E8%A3%85%E8%AE%A1%E7%AE%97%E8%8A%82%E7%82%B9/_compare/f387aa1d91cd2386e6379d8acc04cbcaf6b1212d...8d73071b40d7ee915425a3c52fa96efb64b43fde
